<!DOCTYPE html>
<html>
<body>
<h1>ViReport v0.0.1 &mdash; 2020-03-16</h1>
<p>

</p>
<h2>Input Dataset</h2>
<p>
The analysis was conducted on a dataset containing 571 sequences. The average sequence length was 29799.611, with a standard deviation of 293.817. The earliest sample date was 2019-12-01, the median sample date was 2020-02-25, and the most recent sample date was 2020-03-11. 
</p>

<figure>
<img src="./report_files/figs/input_sequence_lengths.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of input sequence lengths</figcaption>
</figure>

<p>

</p>

<figure>
<img src="./report_files/figs/input_sample_dates.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of input sample dates</figcaption>
</figure>

<p>

</p>
<h2>Preprocessed Dataset</h2>
<p>
The input dataset was preprocessed such that sequences were given safe names: non-letters/digits in sequence IDs were converted to underscores. After preprocessing, the dataset contained 571 sequences. The average sequence length was 29799.611, with a standard deviation of 293.817. The earliest sample date was 2019-12-01, the median sample date was 2020-02-25, and the most recent sample date was 2020-03-11. 
</p>

<figure>
<img src="./report_files/figs/processed_sequence_lengths.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of preprocessed sequence lengths</figcaption>
</figure>

<p>

</p>

<figure>
<img src="./report_files/figs/processed_sample_dates.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of preprocessed sample dates</figcaption>
</figure>

<p>

</p>
<h2>Multiple Sequence Alignment</h2>
<p>
Multiple sequence alignment was performed using MAFFT (Katoh & Standley, 2013) in automatic mode. There were 35056 positions (11805 invariant) and 512 unique sequences in the multiple sequence alignment. Pairwise distances were computed from the multiple sequence alignment using the tn93 tool of HIV-TRACE (Pond et al., 2018). The average pairwise sequence distance was 0.000286, with a standard deviation of 0.000204. 
</p>

<figure>
<img src="./report_files/figs/pairwise_distances_sequences.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of pairwise sequence distances</figcaption>
</figure>

<p>

</p>

<figure>
<img src="./report_files/figs/alignment_entropies.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Shannon entropy across the positions of the multiple sequence alignment</figcaption>
</figure>

<p>

</p>
<h2>Phylogenetic Inference</h2>
<p>
A maximum-likelihood phylogeny was inferred under the General Time-Reversible (GTR) model (Tavare, 1986) using FastTree 2 (Price et al., 2010) using a Gamma20-based likelihood. The inferred phylogeny was MinVar-rooted using FastRoot (Mai et al., 2017). Pairwise distances were computed from the phylogeny using TreeSwift (Moshiri, 2020). The maximum pairwise phylogenetic distance (i.e., tree diameter) was 0.00471, and the average pairwise phylogenetic distance was 0.00052, with a standard deviation of 0.000365. 
</p>

<figure>
<img src="./report_files/figs/tree_mutations.png" width="auto" height="auto" style="max-width:100%;max-height:100%;">
<figcaption>Rooted phylogenetic tree in unit of expected per-site mutations</figcaption>
</figure>

<p>

</p>

<figure>
<img src="./report_files/figs/pairwise_distances_tree.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of pairwise phylogenetic distances</figcaption>
</figure>

<p>

</p>
<h2>Phylogenetic Dating</h2>
<p>
The rooted phylogeny was dated using treedater (Volz & Frost, 2017). The height of the dated tree was 103.983 days, so given that the most recent sample was collected on 2020-03-11, the estimated time of the most recent common ancestor (tMRCA) was 2019-11-28. 
</p>

<figure>
<img src="./report_files/figs/tree_time.png" width="auto" height="auto" style="max-width:100%;max-height:100%;">
<figcaption>Dated phylogenetic tree in unit of years</figcaption>
</figure>

<p>

</p>
<h2>Ancestral Sequence Reconstruction</h2>
<p>
Ancestral sequence reconstruction was performed using TreeTime (Sagulenko et al., 2018). 
</p>
<h2>Transmission Clustering</h2>
<p>
Transmission clustering was performed using TreeN93 (Moshiri, 2018) using pairwise phylogenetic distances. The total number of singletons (i.e., non-clustered individuals) was 92, and the total number of clusters (excluding singletons) was 21. The average cluster size (excluding singletons) was 22.286, with a standard deviation of 47.218, and the maximum and minimum cluster sizes were 211 and 2, respectively. 
</p>

<figure>
<img src="./report_files/figs/cluster_sizes.png" width="auto" height="auto" style="max-width:75%;">
<figcaption>Distribution of cluster sizes (excluding singletons)</figcaption>
</figure>

<p>

</p>
<h2>Citations</h2>
<p>

<ul>
  <li>Katoh K., Standley D.M. (2013). "MAFFT Multiple Sequence Alignment Software Version 7: Improvements in Performance and Usability". Molecular Biology and Evolution. 30(4), 772-780.</li>
  <li>Le S.Q., Gascuel O. (2008). "An Improved General Amino Acid Replacement Matrix". Molecular Biology and Evolution. 25(7), 1307-1320.</li>
  <li>Mai U., Sayyari E., Mirarab S. (2017). "Minimum Variance Rooting of Phylogenetic Trees and Implications for Species Tree Reconstruction". PLoS ONE. 12(8), e0182238.</li>
  <li>Moshiri N. (2018). "TreeN93: a non-parametric distance-based method for inferring viral transmission clusters". bioRxiv.</li>
  <li>Moshiri N. (2020). "TreeSwift: a massively scalable Python tree package". SoftwareX. In press.</li>
  <li>Moshiri N. (2020). "ViReport" (https://github.com/niemasd/ViReport).</li>
  <li>Pond S.L.K., Weaver S., Leigh Brown A.J., Wertheim J.O. (2018). "HIV-TRACE (TRAnsmission Cluster Engine): a Tool for Large Scale Molecular Epidemiology of HIV-1 and Other Rapidly Evolving Pathogens". Molecular Biology and Evolution. 35(7), 1812-1819.</li>
  <li>Price M.N., Dehal P.S., Arkin A.P. (2010). "FastTree 2 -- Approximately Maximum-Likelihood Trees for Large Alignments". PLoS ONE. 5(3), e9490.</li>
  <li>Sagulenko P., Puller V., Neher R.A. (2018). "TreeTime: Maximum-likelihood phylodynamic analysis". Virus Evolution. 4(1), vex042.</li>
  <li>Tavare S. (1986). ""Some Probabilistic and Statistical Problems in the Analysis of DNA Sequences". Lectures on Mathematics in the Life Sciences. 17, 57-86.</li>
  <li>Volz E.M., Frost S.D.W. (2017). "Scalable relaxed clock phylogenetic dating". Virus Evolution. 3(2), vex025.</li>

</ul>

</p>
</body>
</html>
